From: Stanislaw Kardach <stanislaw.kardach@cavium.com>
Date: Wed, 20 Apr 2016 15:05:15 -0700
Subject: [PATCH] Add configuration item for WSREP SST provider

This patch adds a radio button in "OpenStack Services" tab of the
environment settings to configure which WSREP SST provider to use for
MySQL.

Signed-off-by: Stanislaw Kardach <stanislaw.kardach@cavium.com>
---
 nailgun/nailgun/fixtures/openstack.yaml | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/nailgun/nailgun/fixtures/openstack.yaml b/nailgun/nailgun/fixtures/openstack.yaml
index ba25093..e855cf0 100644
--- a/nailgun/nailgun/fixtures/openstack.yaml
+++ b/nailgun/nailgun/fixtures/openstack.yaml
@@ -932,6 +932,27 @@
             description: "Name for Mongo replication set"
             weight: 30
             type: "text"
+        mysql_wsrep:
+          metadata:
+            label: "Database configuration"
+            weight: 15
+            group: "openstack_services"
+          wsrep_method:
+            value: "xtrabackup-v2"
+            label: "WSREP SST provider"
+            description: "This setting changes the wsrep_sst_method switch in the MySQL Galera cluster configuration."
+            weight: 10
+            type: "radio"
+            values:
+              - data: "xtrabackup-v2"
+                label: "xtrabackup-v2"
+                description: "Use Percona xtrabackup v2 provider. This is broken on some ARM64 platforms due to outdated MySQL/InnoDB code used by xtrabackup."
+              - data: "rsync"
+                label: "rsync"
+                description: "Use rsync provider. Use if xtrabackup is broken on your platform."
+              - data: "mysqldump"
+                label: "mysqldump"
+                description: "Use mysqldump provider (untested)."
         additional_components:
           metadata:
             label: "Additional Components"
